ouroboros-consensus/ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Ledger/Block.hs

```diff
@@ -330,8 +331,22 @@ instance MockProtocolSpecific c ext => ApplyTx (SimpleBlock c ext) where
 
   type ApplyTxErr (SimpleBlock c ext) = MockError (SimpleBlock c ext)
 
-  applyTx   = const updateSimpleUTxO
-  reapplyTx = const updateSimpleUTxO
+  applyTx           = reapplyTx
+  reapplyTx _ tx tl =
+      case hasExpired of
```

Comment:
I see CI is failing with a test related to tx/block validation. I believe you also have to take transaction expiration into account when validating blocks (`applyLedgerBlock`). You can move the expiration logic to some common place.

Reply:
It shouldn't live in `applyTx` _or_ `applyLedgerBlock`, but rather in the ledger rules.

Reply:
See Lemma 3 in #1765 :slightly_smiling_face:

Reply:
I pushed it into the ledger rules. I introduced the `Ticked` type used in a few modules; should it be renamed and/or located somewhere more general? I think of it (as commented) as a generalization of `TickedLedgerState`. @edsko @mrBliss